tests/utilities/test_fetching.py

```diff
@@ -263,6 +263,56 @@ def training_epoch_end(self, *_):
     trainer.fit(model)
 
 
+@pytest.mark.parametrize("fn", ("validate", "test"))
+@RunIf(min_torch="1.8.0")
+def test_fetching_dataloader_iter_running_stages(fn, tmpdir):
+    class TestModel(BoringModel):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.automatic_optimization = False
+            self.count = 0
+            self.batches = []
+
+        def training_step(self, dataloader_iter, batch_idx):
+            assert self.count == batch_idx
+            assert isinstance(self.trainer.fit_loop._data_fetcher, DataLoaderIterDataFetcher)
+            # fetch 2 batches
+            self.batches.append(next(dataloader_iter))
+            self.batches.append(next(dataloader_iter))
+
+            batch = self.batches.pop(0)
+            assert isinstance(batch, torch.Tensor) or batch is None
+            self.count += 2
+            opt = self.optimizers()
+            output = self(batch)
+            loss = self.loss(batch, output)
+            opt.zero_grad()
+            loss.backward()
+            opt.step()
```

Comment:
Why override `training_step` if the test does not call `fit`?

And `automatic_optimization`?